lib/matplotlib/artist.py

```diff
@@ -1262,19 +1262,33 @@ def format_cursor_data(self, data):
             method yourself.
 
         The default implementation converts ints and floats and arrays of ints
-        and floats into a comma-separated string enclosed in square brackets.
+        and floats into a comma-separated string enclosed in square brackets,
+        unless the artist has an associated colorbar, in which case scalar
+        values are formatted using the colorbar's formatter.
 
         See Also
         --------
         get_cursor_data
         """
-        try:
-            data[0]
-        except (TypeError, IndexError):
-            data = [data]
-        data_str = ', '.join('{:0.3g}'.format(item) for item in data
-                             if isinstance(item, Number))
-        return "[" + data_str + "]"
+        if np.ndim(data) == 0 and getattr(self, "colorbar", None):
```

Comment:
Is there better criterion than having a colorbar?

You still have a value (~color) in the mesh even if that's not explained via a colorbar. It would be nice if this still works for a simple `plt.pcolormesh()` call. Also if you plot on multiple Axes with "shared" colors AFAIK only one of the Mappables will be associated with the colorbar.

Reply:
The "colorbar-for-value-formatting" discussion is at https://github.com/matplotlib/matplotlib/pull/12473 (I personally just patch matplotlib locally with that PR...).  I think it's orthogonal to the feature here?